### ISSUE TITLE
RT indices not persisted on reboot (TS 3.14)

### ISSUE THREAD

**Issue by al3izz** [pat/thinking-sphinx#986]
I'm running Thinking Sphinx 3.14 in production mode and using Real Time indices. I noticed that when I reboot the server, the index data is not persisted. I end up having to re-index before being able to use the rails app. So, as a solution, I decided to setup a cron job to Flush the indices every 1 minute so that the latest indices would be saved to disk. However, even this does not work, as when I reboot the server, although there are .ram .kill and such files present inside db/sphinx/production/, the rails app still complains of missing indices.

Index data is simply not persisted. How do I solve this?

Here's the example error I get whenever I try to save/delete something:

> FATAL -- :
> ThinkingSphinx::SphinxError (no such index 'person_core'

And it complains like so for the rest of my models.

**Comment by pat** [comment]
Hmm, this is certainly not the regular behaviour!

When you say you're re-indexing, do you mean you're running `rake ts:generate` (or `rake ts:regenerate`)? How many indices have you defined? Are they all real-time indices?

**Comment by al3izz** [comment]
I'm using `rake ts:regenerate`, but I've also tried different options like `rake ts:stop ts:configure ts:start ts:generate` and it still does the same thing when i restart -- the index data does not persist.

I've defined 12 indices, all of which are Real-Time.

**Comment by al3izz** [comment]
i dont know if it helps but I notice that the sphinxsearch service auto starts on boot, and this doesn't allow me to re-index properly, i have to stop the service then index using the rake commands, otherwise it stalls and then fails.

**Comment by pat** [comment]
Just to take a step back (because I've still got no idea what the problem is): if you flush the indices, wait a few seconds, run `rake ts:stop` and then `rake ts:start`, the data is no longer available in Sphinx?

**Comment by al3izz** [comment]
The data is available when I do that, but once I restart, it becomes unavailable, and I now know why:

its because when I restart the computer, the service sphinxsearch automatically starts. And there is no service PID associated with Thinking Sphinx, so for some reason the auto starting doesn't work together.

So I have to run `sudo service sphinxsearch stop`, and then `rake ts:start`, and then it starts working and the flushed data is used.

So, in summary, I have to:
1. flush the indices every 1 minute while the app is in use
2. disable autostart for the sphinxsearch service
3. make a cron job to run `rake ts:start` on every restart

And that solves the issue, thanks.

**Comment by pat** [comment]
Ah, good to know! :)
